### PR TITLE
resource/cloudflare_ruleset: add support for WAF payload logging

### PR DESCRIPTION
**Pull request by jacobbednarz**
Depends on cloudflare/cloudflare-go#701

### CODE REVIEW

File: website/docs/r/ruleset.html.markdown [hunk 1]
```diff
@@ -119,8 +119,14 @@ The following arguments are supported:
 * `products` - (Optional) Products to target with the actions. Valid values are `"bic"`, `"hot"`, `"ratelimit"`, `"securityLevel"`, `"uablock"`, `"waf"` or `"zonelockdown"`.
 * `ruleset` - (Optional) Which ruleset to target. Valid value is `"current"`.
 * `uri` - (Optional) List of URI properties to configure for the ruleset rule when performing URL rewrite transformations (refer to the [nested schema](#nestedblock--action-parameters-uri)).
+* `matched_data` - (Optional) List of properties to configure WAF payload logging (refer to the [nested schema](#nestedblock--action-parameters-matched-data)).
```
Comment (jacobbednarz) [inline website/docs/r/ruleset.html.markdown:122]: cc @pedrosousa

File: cloudflare/resource_cloudflare_ruleset_test.go [hunk 2]
```diff
@@ -978,3 +1021,28 @@ func testAccCheckCloudflareRulesetManagedWAFWithIDBasedOverrides(rnd, name, zone
...
+    description = "%[1]s ruleset description"
+    kind        = "zone"
+    phase       = "http_request_firewall_managed"
+
+    rules {
+      action = "execute"
+      action_parameters {
+        id = "efb7b8c949ac4650a09736fc376e9aee"
+        matched_data {
+					public_key = "not_a_real_public_key"
```
Comment (pedrosousa) [inline cloudflare/resource_cloudflare_ruleset_test.go:1039]: Nit: the indentation looks odd in these two lines (using tabs, unlike the surrounding lines).

File: website/docs/r/ruleset.html.markdown [hunk 3]
```diff
@@ -119,8 +119,14 @@ The following arguments are supported:
...
 * `version` - (Optional)
 
+<a id="nestedblock--action-parameters-matched-data"></a>
+**Nested schema for `matched_data`**
+
+* `public_key` - (Optional) Public key to use within WAF Ruleset payload logging to view the HTTP request parameters.
```
Comment (pedrosousa) [inline website/docs/r/ruleset.html.markdown:128]:
```suggestion
* `public_key` - (Optional) Public key to use within WAF Ruleset payload logging to view the HTTP request parameters. You can generate a public key [using the `matched-data-cli` command-line tool](https://developers.cloudflare.com/waf/managed-rulesets/payload-logging/command-line/generate-key-pair) or [in the Cloudflare dashboard](https://developers.cloudflare.com/waf/managed-rulesets/payload-logging/configure).
```

File: website/docs/r/ruleset.html.markdown (hunk 1 above)
Comment (pedrosousa) [reply website/docs/r/ruleset.html.markdown:122]: Thanks for the heads-up! I've reviewed the content you added and left a few comments.

**Pull request merged**